tools/mksurfdata_esmf/gen_mksurfdata_jobscript_single.py

```diff
@@ -81,14 +93,21 @@ def main ():
         runfile.write(f"#PBS -A {account} \n")
         runfile.write('#PBS -N mksurfdata \n')
         runfile.write('#PBS -j oe \n')
-        runfile.write('#PBS -q regular \n')
         runfile.write('#PBS -l walltime=30:00 \n')
-        runfile.write(f"#PBS -l select={number_of_nodes}:ncpus=36:mpiprocs={tasks_per_node} \n")
+        if machine == 'cheyenne':
+            runfile.write('#PBS -q regular \n')
+            runfile.write(f"#PBS -l select={number_of_nodes}:ncpus=36:mpiprocs={tasks_per_node} \n")
+        elif machine == 'casper':
+            runfile.write('#PBS -q casper \n')
+            runfile.write(f"#PBS -l select={number_of_nodes}:ncpus=12:mpiprocs={tasks_per_node} \n")
         runfile.write("\n")
 
         np = int(tasks_per_node) * int(number_of_nodes)
 
-        output = f"mpiexec_mpt -p \"%g:\" -np {np} ./bld/mksurfdata < {namelist_file}"
+        if machine == 'cheyenne':
+            output = f"mpiexec_mpt -p \"%g:\" -np {np} ./bld/mksurfdata < {namelist_file}"
```

Comment:
@jedwards4b 
does this -p option seem useful to you here for running the mksurfdata_esmf executable? Or should we remove it?

Reply:
The -p option prepends the task id to the stdout file,  I find it useful and would leave it in for cheyenne.   But we should not need to have scripts which are specific to any machine, using cime you should be able to generate a exe command suitable for any platform.

Reply:
Thank you @jedwards4b